docs/kubernetes_setup.md

```diff
@@ -0,0 +1,188 @@
+# Setting up apiserver-network-proxy in Kubernetes
+
+This example will set up apiserver-network-proxy server as a Deployment in the master and apiserver-network-proxy agent as a DaemontSet on the worker nodes.
+
+## Prerequisites
+
+In order to set up the network-proxy, it's needed to generate 2 CA. One for the master communication, `konnectivity-server-ca.crt`
+and one for the cluster communication, `konnectivity-cluster-ca.crt`.
+
+Then, create:
+- A client authentication certificate for the network-proxy agent, `konnectivity-agent-client.crt` and `konnectivity-agent-client.key`.
+- A server authentication certificate for the cluster side of the network-proxy server, `konnectivity-cluster.crt` and `konnectivity-cluster.key` (one for each node or one per node).
+- A client authentication certificate for the Kubernetes API server, `konnectivity-server-client.crt` and `konnectivity-server-client.key`.
+- A server authentication certificate for the server side of the network-proxy server, `konnectivity-server.crt` and `konnectivity-server.key`.
+
+## Master setup
+
+### Kubernetes API server configuration
+
+The Kubernetes API server needs to be started with the following flag: `--egress-selector-config-file=/path/to/your/egress-selector-configuration.yaml`.
+The `egress-selector-configuration.yaml` file looks like:
+
+```yaml
+apiVersion: apiserver.k8s.io/v1alpha1
+kind: EgressSelectorConfiguration
+egressSelections:
+- name: cluster
+  connection:
+    type: http-connect
+    httpConnect:
+      # Here we use 127.0.0.1 since we are binding the apiserver-network-proxy ports on the host
+      # It's possible to use a service instead
+      url: https://127.0.0.1:8131
+      caBundle: /path/to/konnectivity-server-ca.crt
+      clientKey: /path/to/konnectivity-server-client.key
+      clientCert: /path/to/konnectivity-server-client.crt
+- name: master
+  connection:
+    type: direct
+- name: etcd
+  connection:
+    type: direct
+```
+
+### apiserver-network-proxy server
+
+To run the `apiserver-network-proxy` server, use the following Deployment:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    k8s-app: konnectivity-server
+  name: konnectivity-server
+spec:
+  selector:
+    matchLabels:
+      k8s-app: konnectivity-server
+  template:
+    metadata:
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+      labels:
+        k8s-app: konnectivity-server
+    spec:
+      containers:
+      - name: konnectivity-server
+        image: gcr.io/google-containers/proxy-server:v0.0.3
```

Comment:
This container was last uploaded in [July](https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/proxy-server@sha256:d36c064e0267ee453662796356d19adb28efad7531317d41cb1c474f6e030dc9/details?tab=info), so maybe the admin port code wasn't yet present? The livenessProbe fails, so the Pod goes into a crashloopbackoff over and over. Either remove or update the container image to be up-to-date

Reply:
I think I'll wait for official builds, and then update the docs

Reply:
What is ETA on official builds? Where is that progress being discussed? Is there a Dockerfile that can be included in this repo which is used for containerizing the binary?

Reply:
I guess it'll be sometimes next week. I'm currently trying to set it up on k8s.gcr.io. Once this is done, it should be pretty quick. The dockerfile is already in the repo though :)

Reply:
I can spin a build if you want to pick up changes faster but that is not ideal. https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/19 is tracking the official build issue. @Sh4d1 is nice enough to work on that issue.